client/history/client.go

```diff
@@ -517,24 +541,24 @@ func (c *clientImpl) createContext(parent context.Context) (context.Context, con
 }
 
 func (c *clientImpl) getThriftClient(hostPort string) historyserviceclient.Interface {
-	c.thriftCacheLock.RLock()
-	client, ok := c.thriftCache[hostPort]
-	c.thriftCacheLock.RUnlock()
+	c.InquireCacheLock.RLock()
```

Comment:
If you just do:
c.InquireCacheLock.RLock()
defer c.InquireCacheLock.Unlock()
here, you won't have to acquire the lock again and the lock is cleaner?

Reply:
As above, I mistakenly changed the naming and this is not part of my scope. But we can open another issue to optimize the locking mechanism.
One reason I can see that why they did this, is they want to release the lock earlier? But I can see that they did that on purpose to improve the performance, as the trade off, it will sacrify some readability

Reply:
ok got it.